### PR TITLE
Update dimensions metadata when resampling a map

### PR DESCRIPTION
**Pull request by prateekiiest**
Issue #1870 
I updated the 'naxis' values in the resampled map.

### CODE REVIEW

File: sunpy/map/mapbase.py [hunk 1]
```diff
@@ -847,6 +847,8 @@ def resample(self, dimensions, method='linear'):
             new_meta['CD2_2'] *= scale_factor_y
         new_meta['crpix1'] = (dimensions[0].value + 1) / 2.
         new_meta['crpix2'] = (dimensions[1].value + 1) / 2.
+        new_meta['naxis1'] = dimensions[0]
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: FITS headers have the axes reversed, so `naxis1 = dimensions[1]` also it should be `dimensions.value` so it's a float.

File: sunpy/map/mapbase.py [hunk 2]
```diff
@@ -847,6 +847,8 @@ def resample(self, dimensions, method='linear'):
             new_meta['CD2_2'] *= scale_factor_y
         new_meta['crpix1'] = (dimensions[0].value + 1) / 2.
         new_meta['crpix2'] = (dimensions[1].value + 1) / 2.
+        new_meta['naxis1'] = dimensions[1].value
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: you should not add these keys if they do not exist in the original metadata.

Comment (prateekiiest) [inline sunpy/map/mapbase.py]: I have send a new PR for the test case for Generic Map #2011

Comment (prateekiiest) [inline sunpy/map/mapbase.py]: Generic Map testing one??

Comment (prateekiiest) [inline sunpy/map/mapbase.py]: def test_resample_metadata(generic_map, sample_method, new_dimensions):
    """
    Check that the resampled map has correctly adjusted metadata.
    """
    resampled_map = generic_map.resample(new_dimensions, method=sample_method)
    assert float(resampled_map.meta['cdelt1']) / generic_map.meta['cdelt1'] \
        == float(generic_map.data.shape[1]) / resampled_map.data.shape[1]
    assert float(resampled_map.meta['cdelt2']) / generic_map.meta['cdelt2'] \
        == float(generic_map.data.shape[0]) / resampled_map.data.shape[0]
    assert resampled_map.meta['crpix1'] == (resampled_map.data.shape[1] + 1) / 2.
    assert resampled_map.meta['crpix2'] == (resampled_map.data.shape[0] + 1) / 2.
    assert resampled_map.meta['crval1'] == generic_map.center.x.value
    assert resampled_map.meta['crval2'] == generic_map.center.y.value
    for key in generic_map.meta:
        if key not in ('cdelt1', 'cdelt2', 'crpix1', 'crpix2',
                       'crval1', 'crval2'):
            assert resampled_map.meta[key] == generic_map.meta[key]

Actually in this part the (if key is not) didn't include the naxis keys. So I just added this naxis1 and naxis2 in that section

**Comment by prateekiiest** [comment]
@Cadair 
Why sphinx documentation warning is issued?? Same occurred in #2011

File: sunpy/map/mapbase.py (hunk 2 above)
Comment (Cadair) [inline sunpy/map/mapbase.py]: My point is that this test should not need these keys. There is no need for us to add the `naxis` keys to the header, only update them if they are already there.

Comment (prateekiiest) [inline sunpy/map/mapbase.py]: Ok , got it.

**Pull request closed without merging**